### PR TITLE
Cut(eos_cli_config_gen): Remove the deprecated keys for vlan_interfaces 

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Remove the deprecated keys for vlan_interfaces 

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Remove the deprecated keys for vlan_interfaces 

## How to test
Run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2 [hunk 1]
```diff
@@ -74,23 +74,19 @@
...
+{%             set row_vrrp = vlan_interface.vrrp_ids.ipv6.address | arista.avd.default('-') %}
 {%             set row_nd_ra_disabled = vlan_interface.ipv6_nd_ra_disabled | arista.avd.default('-') %}
 {%             set row_nd_man_cfg = vlan_interface.ipv6_nd_managed_config_flag | arista.avd.default('-') %}
 {%             set row_nd_oth_cfg = vlan_interface.ipv6_nd_other_config_flag | arista.avd.default('-') %}
 {%             set row_acl_in = vlan_interface.ipv6_access_group_in | arista.avd.default('-') %}
 {%             set row_acl_out = vlan_interface.ipv6_access_group_out | arista.avd.default('-') %}
-| {{ vlan_interface.name }} | {{ row_vrf }} | {{ row_ip_addr }} | {{ row_ip_vaddr }} | {{ row_varp }} | {{ row_vrrp }} | {{ row_nd_ra_disabled }} | {{ row_nd_man_cfg }} | {{ row_nd_oth_cfg }} | {{ row_acl_in }} | {{ row_acl_out }} |
+| {{ vlan_interface.name }} | {{ row_vrf }} | {{ row_ip_addr }} | {{ row_ip_vaddr | arista.avd.default('-') }} | {{ row_varp | arista.avd.default('-') }} | {{ row_vrrp }} | {{ row_nd_ra_disabled }} | {{ row_nd_man_cfg }} | {{ row_nd_oth_cfg }} | {{ row_acl_in }} | {{ row_acl_out }} |
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2]: Keep the default in the previous lines

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml [hunk 2]
```diff
@@ -22,25 +19,15 @@ vlan_interfaces:
...
-    # ipv6_address_virtual deprecated in 4.0.0. To be removed in 5.0.0
-    ipv6_address_virtual: fc00:10:10:2::1/64
-    ipv6_address_virtuals:
-      - fc00:10:11:2::1/64
-      - fc00:10:12:2::1/64
 
   Vlan3:
     description: test ipv6_address_virtual
     ipv6_enable: true
     ipv6_address: 1b11:3a00:22b3:5200::15/64
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml]: I don't think we need Vlan3 anymore

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml [hunk 3]
```diff
@@ -6,9 +6,6 @@ vlan_interfaces:
   Vlan1:
     description: test ipv6_address_virtual
     ipv6_enable: true
-    # Testing ipv6_address_virtual alone
-    # ipv6_address_virtual deprecated in 4.0.0. To be removed in 5.0.0
-    ipv6_address_virtual: fc00:10:10:1::1/64
 
   Vlan2:
     description: test ipv6_address_virtual and ipv6_address_virtuals
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml]: change the description to what this tests is now doing

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml [hunk 4]
```diff
@@ -6,9 +6,6 @@ vlan_interfaces:
   Vlan1:
     description: test ipv6_address_virtual
     ipv6_enable: true
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml]: I don't think we need Vlan1 anymore

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2 (hunk 1 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2]: updated

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml (hunk 2 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml]: removed

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml (hunk 3 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml]: updated

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml (hunk 4 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/vlan-interfaces.yml]: removed

**Review by gmuloc — APPROVED** [review]

**Comment by laxmikantchintakindi** [comment]
Minor change:

https://github.com/aristanetworks/avd/pull/4305/files#diff-3df6ba1062882cc65140a880885ef08ba244baff1f1f9f4a76ad0cd08c23e6c3L72

Virtual Router Address -> `Virtual Router Addresses`

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2 [hunk 5]
```diff
@@ -50,7 +50,7 @@
 {%         set row_ip_addr = vlan_interface.ip_address | arista.avd.default('-') %}
 {%         set row_ip_vaddr = vlan_interface.ip_address_virtual | arista.avd.default('-') %}
 {%         set row_varp = vlan_interface.ip_virtual_router_addresses | arista.avd.default('-') | join(", ") %}
-{%         set row_vrrp = vlan_interface.vrrp.ipv4 | arista.avd.default('-') %}
+{%         set row_vrrp = vlan_interface.vrrp_ids.ipv4.address | arista.avd.default('-') %}
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2]: vrrp_ids is a list. AVD will throw error for this line.

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2 [hunk 6]
```diff
@@ -74,17 +74,9 @@
...
-{%             endif %}
-{%             if vlan_interface.ipv6_virtual_router_addresses is arista.avd.defined %}
-{%                 set row_varp = vlan_interface.ipv6_virtual_router_addresses | join(", ") %}
-{%             else %}
-{%                 set row_varp = vlan_interface.ipv6_virtual_router_address | arista.avd.default('-') %}
-{%             endif %}
-{%             set row_vrrp = vlan_interface.vrrp.ipv6 | arista.avd.default('-') %}
+{%             set row_ip_vaddr = vlan_interface.ipv6_address_virtuals | arista.avd.default('-') | join(", ") %}
+{%             set row_varp = vlan_interface.ipv6_virtual_router_addresses | arista.avd.default('-') | join(", ") %}
+{%             set row_vrrp = vlan_interface.vrrp_ids.ipv6.address | arista.avd.default('-') %}
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2]: vrrp_ids is a list. AVD will throw error for this line.

Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2]: We can remove this column as it gets displayed in `VRRP details` table.

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2 (hunk 5 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vlan-interfaces.j2]: removed

**Review by laxmikantchintakindi — APPROVED** [review]

**Review by gmuloc — APPROVED** [review]

**Pull request merged**